packages/grid/x-data-grid-pro/src/hooks/features/rowGrouping/useGridRowGrouping.tsx

```diff
@@ -351,6 +151,30 @@ export const useGridRowGrouping = (
   /**
    * PRE-PROCESSING
    */
+  const addColumnMenuButtons = React.useCallback<GridPreProcessor<'columnMenu'>>(
```

Comment:
Just a copy paste, thie processing of this files were in two parts of the hook.

Maybe we could move all of them to the `useGridRowGrouping` hook.
For those recent feature hooks; not a lot would remain (see `useGridTreeData` that has a single function now).